### PR TITLE
Bugfix: `run_metadata` value returns string instead of other types

### PR DESCRIPTION
**Pull request by avishniakov**
## Describe changes
I fixed RunMetadata models to properly handle primitive types of metadata and not dump all of them into a string format.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: tests/integration/functional/artifacts/test_utils.py [hunk 1]
```diff
@@ -124,6 +124,11 @@ def test_log_artifact_metadata_existing(clean_client):
         artifact_name="meaning_of_life",
         artifact_version="1",
     )
+    log_artifact_metadata(
```
Comment (htahir1) [inline tests/integration/functional/artifacts/test_utils.py:127]: can you also tests list of strings and floats to make sure?

**Review by htahir1 — COMMENTED** [review]
One short thing on my side. Also, are we sure that this Union property wont bite us in other ways?

File: tests/integration/functional/artifacts/test_utils.py (hunk 1 above)
Comment (avishniakov) [reply tests/integration/functional/artifacts/test_utils.py:127]: Extended

**Review by bcdurak — APPROVED** [review]
The tests seem to prove the point behind this PR and the changes look good to me. 

Still, I am also not 100% sure if this would have any cascading effects if we were to add the same behaviour to our older models. It is also fully clear on their docs, how `smart`, the `smart_union` is and they even have a warning for compound types. 

As long as the tests are succeeding, I think we can merge this for the `run_metadata` though.

**Pull request merged**